RecoMuon/MuonIdentification/plugins/MuonIdProducer.cc

```diff
@@ -984,11 +988,17 @@ void MuonIdProducer::fillArbitrationInfo( reco::MuonCollection* pOutputMuons )
      // chamberIter1
      for ( auto& chamber1 : muon1.matches() )
      {
-       if(chamber1.segmentMatches.empty()) continue;
+       // segmentIter1
+       std::vector<reco::MuonSegmentMatch> * segmentMatches1 = nullptr;
+       if      (muonType == reco::Muon::TrackerMuon) segmentMatches1 =  & chamber1.segmentMatches;
```

Comment:
Since this pattern of logic is repeated three times in the code, it might be preferable to set up a helper function for it. (One could even get fancy with a non-type template parameter, but it's not really necessary.) Also, then the exception would be thrown consistently (right now it's only thrown in the first instance, not in the other instances).